test/fixtures/gcs-fixture/src/main/java/fixture/gcs/GoogleCloudStorageHttpHandler.java

```diff
@@ -308,10 +295,7 @@ private String httpServerUrl(final HttpExchange exchange) {
             } else {
                 // removes the trailing end "\r\n--__END_OF_PART__--\r\n" which is 23 bytes long
                 int len = fullRequestBody.length() - startPos - 23;
-                final InputStream stream = fullRequestBody.slice(startPos, len).streamInput();
-                final byte[] buffer = new byte[len];
-                Streams.readFully(stream, buffer);
-                content = Tuple.tuple(name, new BytesArray(buffer));
+                content = Tuple.tuple(name, fullRequestBody.slice(startPos, len));
```

Comment:
Even though `slice` wastes some memory here for the wrapping around the actual blob content for smaller blobs, it's a huge efficiency gain to stick with `16kb` chunks from the `BytesReference` compared to allocating a potentially MB sized array.